views/js/qtiCreator/widgets/interactions/mediaInteraction/states/Question.js

```diff
@@ -20,153 +44,210 @@ define([
 
     var MediaInteractionStateQuestion = stateFactory.extend(Question, initQuestionState, exitQuestionState);
 
-    MediaInteractionStateQuestion.prototype.initForm = function(){
-
-        var widget = this.widget;
-        var $form = widget.$form;
-        var $container = widget.$original;
-        var options = widget.options;
-        var interaction = widget.element;
-        var callbacks = {};
+    /**
+     * Initialize the attribute form : file, type, size, etc.
+     */
+    MediaInteractionStateQuestion.prototype.initForm = function initForm(){
+
+        var widget             = this.widget;
+        var $form              = widget.$form;
+        var $container         = widget.$original;
+        var options            = widget.options;
+        var interaction        = widget.element;
+        var isAudio            = false;
+        var defaultVideoHeight = 270;
+        var defaultAudioHeight = 30;
+        var callbacks;
+        var $heightContainer;
+
+        /**
+         * Each change triggers an re rendering of the interaction
+         */
+        var reRender = _.debounce(function reRender(){
+            interaction.attr('responseIdentifier', interaction.attr('responseIdentifier'));
 
-        var reRender = _.throttle(function reRender(interaction){
-            xmlUpdateCheat(interaction);
             widget.destroyInteraction();
             widget.renderInteraction();
         }, 1000);
 
-        var xmlUpdateCheat = function xmlUpdateCheat(interaction){
-            // xml update cheat
-            interaction.attr('responseIdentifier', interaction.attr('responseIdentifier'));
+        /**
+         * Switch to audio mode:
+         * update height and disable the field
+         */
+        var switchToAudio = function switchToAudio(){
+            isAudio = true;
+
+            $heightContainer.hide();
+            interaction.object.attr('height', defaultAudioHeight);
+        };
+
+        /**
+         * Switch to video mode:
+         * update height and enable the field
+         */
+        var switchToVideo = function switchToVideo(){
+            isAudio = false;
+            interaction.object.attr('height', defaultVideoHeight);
+            $heightContainer.show();
+        };
+
+        /**
+         * Switch mode based on file type
+         */
+        var switchMode = function switchMode(){
+            if (/audio/.test(interaction.object.attr('type'))) {
```

Comment:
Does not work with OGG format. The MIME type can be the same for both audio and video OGG (`application/ogg`). I am wondering if this can be an issue as this format is not supported by all browsers.